### DISCUSSION TITLE
Maui controls in material control

### DISCUSSION THREAD

**Discussion by sly13010** [enisn/UraniumUI#585] [Q&A]
hi!

if in a xaml file i have for exemple :

```
<material:TreeView 
    ItemsSource="{Binding Noeuds}"
    ChildrenBinding="{Binding Noeuds}"
    SelectionMode="Single"
    SelectedItem="{Binding SelectedItem}"
    x:Name="mytree">
    <material:TreeView.ItemTemplate>
        <DataTemplate>
            <HorizontalStackLayout Spacing="5">
                <Image Source="{FontImageSource FontFamily=MaterialRegular, Glyph={x:Static m:MaterialRegular.Folder}, Color={Binding Action, Converter={StaticResource ConvertActionToStyle}}}" />
                <Label Text="{Binding Name}" FontAttributes="Bold" x:Name="test1"/>
                <Label Text="{Binding Noeuds.Count, StringFormat='({0})'}" />
                <Button Text="test" Clicked="Button_Clicked" x:Name="test2"/>
            </HorizontalStackLayout>
        </DataTemplate>
    </material:TreeView.ItemTemplate>
</material:TreeView>
```
and i want to access my maui control button in the event i don't find le controle by is name

```
private void Button_Clicked(object sender, EventArgs e)
{
  test2. // does not exist 
}
```

How to access to test2 ? the trouble is only when a maui control is in a material control

Best regards

**Comment by enisn** [comment]
There will be more than one Button when you bind Items to the TreeView. If you really want to execute something **for each** item in the TreeView, you can create a new component by creating a new contentview file and write your x:Name and the logic in it and use that component inside DataTemplate.

In your condition, it seems you need to access sender, so you can use sender as button

if(sender is Button myButton)
{
    myButton.Text = "Clicked";
}
